From 8114d1faee7ef8a7eef06bd8056db7bc2f34419e Mon Sep 17 00:00:00 2001
From: Bin Han <bin.b.han@intel.com>
Date: Wed, 20 Jan 2016 14:05:13 +0800
Subject: [PATCH 04/13] atomisp2: always turn off DZ for ISP2.7

ISP2.7 does not really support digital zoom, turn it off
by default to avoid fails at constructing the pipeline.

Original-Tracked_On: #H1504150170
Change-Id: I57a746bb6b924c0971f637aac16865e691889f02
Signed-off-by: Bin Han <bin.b.han@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/96880
Validation-Android: Wei Tao <wei.tao@intel.com>
Reviewed-by: HaiqiangX Yao <haiqiangx.yao@intel.com>
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Validation-Android: Dessert M CHT FFD (SC)
Approver: Tian Shu Qiu <tian.shu.qiu@intel.com>
Reviewed-by: GuoqingX Zhang <guoqingx.zhang@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-17482
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/464447
---
 .../media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c      | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
index e4ec9aa..db33f502 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_compat_css20.c
@@ -723,12 +723,16 @@ static void __apply_additional_pipe_config(
 		 * control whether use isp2.7 */
 #ifdef V4L2_CID_ATOMISP_SELECT_ISP_VERSION
 		if (asd->select_isp_version->val ==
-			ATOMISP_CSS_ISP_PIPE_VERSION_2_7)
+			ATOMISP_CSS_ISP_PIPE_VERSION_2_7) {
 			stream_env->pipe_configs[pipe_id].isp_pipe_version =
 				SH_CSS_ISP_PIPE_VERSION_2_7;
-		else
+			/* ISP2.7 does not support DZ, we should always turn
+			   it off. */
+			stream_env->pipe_configs[pipe_id].enable_dz = false;
+		} else {
 			stream_env->pipe_configs[pipe_id].isp_pipe_version =
 				SH_CSS_ISP_PIPE_VERSION_2_2;
+		}
 #endif
 		break;
 	case IA_CSS_PIPE_ID_VIDEO:
-- 
1.9.1

